src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -756,6 +762,13 @@ def _init_arch_params(self) -> None:
         if arch_params is not None:
             self.arch_params.override(**arch_params.to_dict())
 
+    def _if_need_to_calc_validation(self, epoch: int) -> bool:
```

Comment:
May I suggest a (hopefully) better name for this function: `_should_run_validation_for_epoch`?